.travis.yml

```diff
@@ -157,8 +157,7 @@ matrix:
             -DPYTHON_VERSION=2
             -DCMAKE_INSTALL_PREFIX=$HOME/install/syslog-ng
             ..
-        - make --keep-going -j $(nproc) all install
-        - ctest -j $(nproc) --output-on-failure
+        - make --keep-going -j $(nproc) all check install
```

Comment:
> 1) only build test programs in case we are running tests, and to couple
   rebuilding of the test programs with running them.

I could agree that there are use cases where one do not want to compile UT, but I know only those cases where also the user do not want to execute them either. (completely disabling the tests are fine and there is a cmake way of doing that: `BUILD_TESTING`).
Please if you have other use case than that please share.
Also the link mention something like this: `This is IMHO a very nice feature, because the normal build is not bogged down by the compilation of many test programs, or worse, comes to a grinding halt due to a compilation error in one of the test programs you didn't have time to fix yet. `
Well, I hope people does not feel "bogged" about not compiling test ...

> 2) to reference the unit test phase as a DEPENDS in add_custom_target(), as
   the "test" target is special, whereas "check" is a normal target.

This is the same reason as (1).

> 3) to remove pep8/pylint from unit test execution in a subsequent patch

Those can be removed with the exact same way as in your current branch, without this one. (they are not really dependent)

The part of the patch that introduces a `make check` that triggers a build and a test execution I am somewhat okay (if there are really people who would like to use that way). But not the part that disables the `test` target built into cmake, and the default test compilation.

Reply:
we've agreed that we would keep the building of test programs in "make all", while also triggering the build of test programs in "make check".